USF-1670: Dropins Localization (#95)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/cart/cart-dictionary.mdx

---

> title: Cart Dictionary

Reviewer: ```suggestion
title: Cart dictionary
```

> description: Reference the default dictionary for the Cart dropin.

Reviewer: ```suggestion
description: Learn how to customize the default dictionary for the cart drop-in component.
```

---

import CodeInclude from '@components/CodeInclude.astro';
import CodeImport from '@components/CodeImport.astro';
import dictionary from '@dropins/storefront-cart/i18n/en_US.json.d.ts?raw';

> The default Cart dictionary file (`i18n/en_US.json`) contains the default values for all labels and text common to the Cart drop-in.

Reviewer: ```suggestion
The default cart dictionary file (`i18n/en_US.json`) contains the default values for all labels and text common to the cart drop-in component.
```

## Default keys and values

<CodeInclude code={dictionary} title="en_US.json.d.ts" lang="ts" />

## Customizing the dictionary

To customize the dictionary, you can create a new dictionary file and import it into your project. The new dictionary file should contain only the keys you want to override. The keys in the new dictionary file will be merged with the default dictionary.

For example, to customize the "Continue Shopping" button text, create a new dictionary file with the following content:

```typescript
// src/i18n/en_US.json.d.ts
export default {
  cart: {
    continueShopping: 'Back to shopping',
  },
};
```

Then import the new dictionary file into your project:

```typescript
import customDictionary from '../i18n/en_US.json.d.ts';

const dictionary = {
  ...defaultDictionary,
  ...customDictionary,
};
```

You can then use the `dictionary` object in your components to access the customized values.

## Dictionary keys

> The following keys are available in the Cart dictionary:

Reviewer: ```suggestion
The following keys are available in the cart dictionary:
```

- `cart`
  - `title`
  - `emptyCart`
  - `continueShopping`
  - `checkout`
  - `subtotal`
  - `shipping`
  - `tax`
  - `total`
  - `quantity`
  - `remove`
  - `undo`
  - `undoRemove`
  - `undoRemoveAll`
  - `undoRemoveAllItems`
  - `undoRemoveItem`
  - `undoRemoveItemSingular`
  - `undoRemoveItems`
  - `undoRemoveItemsPlural`
  - `undoRemoveItemsSingular`
  - `undoRemoveItemsSingularPlural`

>   - `undoRemoveItems

Reviewer: ```suggestion
  - `undoRemoveItems`
```
